packages/expo/src/hooks/useLocalCredentials/useLocalCredentials.ts

```diff
@@ -0,0 +1,186 @@
+        numericTypes.includes(AuthenticationType.FACIAL_RECOGNITION)
+      ) {
+        setAuthenticationType('face-recognition');
+      } else {
+        setAuthenticationType('fingerprint');
+      }
+    });
+
+    return () => {
+      ignore = true;
+    };
+  }, []);
+
+  return authenticationType;
+};
+
+const useUserOwnsCredentials = ({ storeKey }: { storeKey: string }) => {
+  const { user } = useUser();
+  const [userOwnsCredentials, setUserOwnsCredentials] = useState(false);
+
+  const getUserCredentials = (storedIdentifier: string | null): boolean => {
+    if (!user || !storedIdentifier) {
+      return false;
+    }
+
+    const identifiers = [
+      user.emailAddresses.map(e => e.emailAddress),
+      user.phoneNumbers.map(p => p.phoneNumber),
+    ].flat();
+
+    if (user.username) {
+      identifiers.push(user.username);
+    }
+    return identifiers.includes(storedIdentifier);
+  };
+
+  useEffect(() => {
+    let ignore = false;
+    getItemAsync(storeKey)
+      .catch(() => null)
+      .then(res => {
+        if (ignore) {
+          return;
+        }
+        setUserOwnsCredentials(getUserCredentials(res));
+      });
+
+    return () => {
+      ignore = true;
+    };
+  }, [storeKey, user]);
+
+  return [userOwnsCredentials, setUserOwnsCredentials] as const;
+};
+
+export const useLocalCredentials = (): LocalCredentialsReturn => {
+  const { isLoaded, signIn } = useSignIn();
+  const { publishableKey } = useClerk();
+
+  const key = `__clerk_local_auth_${publishableKey}_identifier`;
+  const pkey = `__clerk_local_auth_${publishableKey}_password`;
+  const [hasLocalAuthCredentials, setHasLocalAuthCredentials] = useState(!!getItem(key));
+  const [userOwnsCredentials, setUserOwnsCredentials] = useUserOwnsCredentials({ storeKey: key });
+  const hasEnrolledBiometric = useEnrolledBiometric();
+  const authenticationType = useAuthenticationType();
+
+  const biometryType = hasEnrolledBiometric ? authenticationType : null;
+
+  const setCredentials = async (creds: LocalCredentials) => {
+    if (!(await isEnrolledAsync())) {
+      return;
+    }
+
+    if (creds.identifier && !creds.password) {
+      return errorThrower.throw(
+        `useLocalCredentials: setCredentials() A password is required when specifying an identifier.`,
+      );
+    }
+
+    if (creds.identifier) {
+      await setItemAsync(key, creds.identifier);
+    }
+
+    const storedIdentifier = await getItemAsync(key).catch(() => null);
+
+    if (!storedIdentifier) {
+      return errorThrower.throw(
+        `useLocalCredentials: setCredentials() an identifier should already be set in order to update its password.`,
+      );
+    }
+
+    setHasLocalAuthCredentials(true);
+    await setItemAsync(pkey, creds.password, {
+      keychainAccessible: WHEN_PASSCODE_SET_THIS_DEVICE_ONLY,
+      requireAuthentication: true,
+    });
+  };
+
+  const clearCredentials = async () => {
+    await Promise.all([deleteItemAsync(key), deleteItemAsync(pkey)]);
```

Comment:
Do we need to handle errors here?

Reply:
`deleteItemAsync` can throw if the value couldn't be deleted. And this is expected. do you think we should catch it and throw our own errors ?

If yes, then we should do this to other places where we also call `setItemAsync`.

Reply:
Makes sense! I don't think we need to handle specifically then.